plugins/modules/remote_management/ipmi/ipmi_power.py

```diff
@@ -133,18 +224,54 @@ def main():
         )
         module.debug('ipmi instantiated - name: "%s"' % name)
 
-        current = ipmi_cmd.get_power()
-        if current['powerstate'] != state:
-            response = {'powerstate': state} if module.check_mode else ipmi_cmd.set_power(state, wait=timeout)
-            changed = True
+        changed = False
+        if not machine:
+            current = ipmi_cmd.get_power()
+            if current['powerstate'] != state:
+                response = {'powerstate': state} if module.check_mode \
+                    else ipmi_cmd.set_power(state, wait=timeout)
+                changed = True
+            else:
+                response = current
+
+            if 'error' in response:
+                module.fail_json(msg=response['error'])
+
+            module.exit_json(changed=changed, **response)
         else:
-            response = current
-            changed = False
+            response = []
+            for entry in machine:
+                taddr = entry['targetAddress']
+                if taddr >= INVALID_TARGET_ADDRESS:
+                    module.fail_json(msg="targetAddress should be set between 0 to 255.")
+
+                try:
+                    # bridge_request is supported on pyghmi 1.5.30 and later
+                    current = ipmi_cmd.get_power(bridge_request={"addr": taddr})
+                except TypeError:
+                    module.fail_json(msg="targetAddress isn't supported on this module")
```

Comment:
You should be more explicit what `module` means. It surely does not mean Ansible module, since this Ansible module supports the `targetAddress` option.